Hippo.Web/Handlers/VerifyRoleAccessHandler.cs

```diff
@@ -0,0 +1,49 @@
+﻿using Hippo.Core.Data;
+using Hippo.Core.Models;
+using Hippo.Core.Services;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.EntityFrameworkCore;
+using System.Security.Claims;
+
+namespace Hippo.Web.Handlers
+{
+    public class VerifyRoleAccessHandler : AuthorizationHandler<VerifyRoleAccess>
+    {
+        private readonly AppDbContext _dbContext;
+
+        private readonly IHttpContextAccessor _httpContext;
+
+        public VerifyRoleAccessHandler(AppDbContext dbContext, IHttpContextAccessor httpContext)
+        {
+            _dbContext = dbContext;
+            _httpContext = httpContext;
+        }
+        protected override async Task HandleRequirementAsync(AuthorizationHandlerContext context, VerifyRoleAccess requirement)
+        {
+            var userIamId = context.User.Claims.SingleOrDefault(c => c.Type == UserService.IamIdClaimType)?.Value;
+            var kerbId = context.User.Claims.SingleOrDefault(a => a.Type == ClaimTypes.NameIdentifier)?.Value;
+
+
+            if (string.IsNullOrWhiteSpace(userIamId))
+            {
+                return;
+            }
+
+            var systemUsers = new[] { "jsylvest", "postit", "cydoval", "sweber" };

```

Comment:
I forget. Did we decide that creating a `Roles` table would be overkill? At least for the first version?

Reply:
@sprucely I think that was what Scott said.